echion/monkey/gevent.py

```diff
@@ -0,0 +1,204 @@
+import typing as t
+from types import FrameType
+
+import gevent
+import gevent.greenlet
+import gevent.hub
+from gevent import thread
+from gevent.greenlet import Greenlet as _Greenlet
+from greenlet import greenlet, settrace
+
+import echion.core as echion
+
+
+# Original objects
+_gevent_hub_spawn_raw = gevent.hub.spawn_raw
+_gevent_joinall = gevent.joinall
+_gevent_wait = gevent.wait
+_gevent_iwait = gevent.iwait
+
+# Global package state
+_greenlet_frames: t.Dict[int, t.Union[FrameType, bool, None]] = {}
+_original_greenlet_tracer: t.Optional[t.Callable[[str, t.Any], None]] = None
+_greenlet_parent_map: t.Dict[int, int] = {}
+_parent_greenlet_count: t.Dict[int, int] = {}
+
+FRAME_NOT_SET = False  # Sentinel for when the frame is not set
+
+
+def track_gevent_greenlet(greenlet: _Greenlet) -> _Greenlet:
+    greenlet_id = thread.get_ident(greenlet)
+    frame: t.Union[FrameType, bool, None] = FRAME_NOT_SET
+
+    echion.track_greenlet(
+        greenlet_id, greenlet.name or type(greenlet).__qualname__, frame
+    )
+
+    # Untrack on completion
+    try:
+        greenlet.rawlink(untrack_greenlet)
+    except AttributeError:
+        # This greenlet cannot be linked (e.g. the Hub)
+        pass
+
+    _greenlet_frames[greenlet_id] = frame
+
+    return greenlet
+
+
+def update_greenlet_frame(
+    greenlet_id: int, frame: t.Union[FrameType, bool, None]
+) -> None:
+    _greenlet_frames[greenlet_id] = frame
+    echion.update_greenlet_frame(greenlet_id, frame)
+
+
+def greenlet_tracer(event: str, args: t.Any) -> None:
+    if event in {"switch", "throw"}:
+        # This tracer function runs in the context of the target
+        origin, target = t.cast(t.Tuple[Greenlet, Greenlet], args)
+
+        if (origin_id := thread.get_ident(origin)) not in _greenlet_frames:
+            try:
+                track_gevent_greenlet(origin)
+            except Exception:
```

Comment:
Do you mind changing this to `AttributeError`?
And the one in line 72 too. 
Do you expect any other exception to be thrown from `track_gevent_greenlet()`?